include/uk/plat/memory.h

```diff
@@ -252,6 +257,61 @@ void *ukplat_memregion_alloc(__sz size, int type, __u16 flags);
  */
 int ukplat_mem_init(void);
 
+/* Allocates and returns an auxiliary stack that can be used in emergency cases
+ * such as when switching system call stacks. The size is that given by
+ * CONFIG_AUXSP_SIZE.
+ *
+ * @param a
+ *   The allocator to use for the auxiliary stack
+ * @param vas
+ *   The virtual address space to use for the mapping of the auxiliary stack.
+ *   This should be used in conjunction with CONFIG_LIBUKVMEM to ensure that
+ *   accesses to the auxiliary stack do not generate page faults in more
+ *   fragile system states.
+ * @param auxsp_len
+ *   The custom length of the auxiliary stack. If 0, then
+ *   CONFIG_UKPLAT_AUXSP_SIZE is used instead as the default length.
+ *
+ * @return
+ *   Pointer to the allocated auxiliary stack
+ */
+static inline __uptr ukplat_auxsp_alloc(struct uk_alloc *a,
```

Comment:
I don't know about the TLS/stack. But the auxiliary stack is definetely something you would use in emergency cases or more fragile situations when a page fault should be avoided. I would say they are different in that regard IMO. As long as you don't have TLS accesses in a page fault handler that is...

Reply:
But now that you bring this up, you reminded me of something that I thought off some time ago, but I forgot. Can we guarantee that there will be no TLS accessses in IRQ/trap handlers? Because if not then, especially for binompat, we will have to be able to store/restore TLS pointer so as not to contaminate the application's TLS.

Reply:
> I don't know about the TLS/stack. But the auxiliary stack is definetely something you would use in emergency cases or more fragile situations when a page fault should be avoided. I would say they are different in that regard IMO. As long as you don't have TLS accesses in a page fault handler that is...

To clarify after internal discussion with @mogasergiu: This is regarding uses of the aux stack for cases other than syscalls. In particular in native mode where the aux stack is not used for syscalls, madvise ensures that we avoid the case where the moment the aux stack is needed it has not been mapped to an actual page yet and will generate a page fault we are unable to handle.

> But now that you bring this up, you reminded me of something that I thought off some time ago, but I forgot. Can we guarantee that there will be no TLS accessses in IRQ/trap handlers? Because if not then, especially for binompat, we will have to be able to store/restore TLS pointer so as not to contaminate the application's TLS.

Given that IRQ handling is decoupled from the currently running thread, it sounds like a reasonable assumption to me that the handler won't need to access thread-specific information.